Flesh out the Download Data button

ianhi opened the pull request:
Built on top of #26 so I could see the javascript and CSS changes locally.

Closes: #16
Closes: #25 

This no longer copies the HTTPS/S3 link. But instead opens a modal with different options for download and makes it clear how to construct an S3 compatible URL


<img width="1173" height="800" alt="image" src="https://github.com/user-attachments/assets/32c752e2-9279-488f-a9fd-657890abe833" />


and here is the button click feedback:


https://github.com/user-attachments/assets/aee21689-da5e-4cdd-946b-bcfe2101702f


Going to github ping @will-moore because I see you've made changes and issues in this area but are not watching the repo. Just wanted to let you know that this exists

ianhi committed:
improve download instructions - add zarr and ome-zarr-py and break up aws cli

will-moore commented:
```
$ echo $TMPDIR 
/var/folders/zz/zyxvpxvq6csfxvn_n0000000000000/T/

$ curl -fsSL https://pixi.sh/install.sh | sh
This script will automatically download and install Pixi (latest) for you.
Getting it from this url: https://github.com/prefix-dev/pixi/releases/latest/download/pixi-aarch64-apple-darwin.tar.gz
mktemp: mkstemp failed on /var/folders/zz/zyxvpxvq6csfxvn_n0000000000000/T//.pixi_install.OGyooWFE: Permission denied
```

I'm not sure how to replicate the `$ pixi install` error, now that everything is already installed, but it was similar.

will-moore commented:
I'm not sure I understand https://github.com/zarr-developers/zarr-python/pull/3496.
It may be correct that "https:" URLs are not "S3" URLs, so the naming needs fixed.

But this works OK:

```
import zarr
z = zarr.open("https://uk1s3.embassy.ebi.ac.uk/idr/zarr/v0.5/idr0062A/6001240_labels.zarr")
print(dict(z.attrs))
```

will-moore commented:
Ah, OK - I understand a bit better now. I've never actually used `group.keys()` because I'm often dealing with web apps where we don't expect to be able to list keys from http URLs. So OME-Zarr doesn't assume that you can list keys, and I've always assumed you wouldn't be able to.
Interesting to see that you can anyway.

In your example, I see that you get a warning when you do this at the end:

```
print(list(z.keys()))

/Users/wmoore/opt/anaconda3/envs/omeroweb_zarrv3/lib/python3.12/site-packages/zarr/core/group.py:3390: ZarrUserWarning: Object at 6001240_labels.zarr is not recognized as a component of a Zarr hierarchy.
  warnings.warn(
['0', '1', 'labels', '2']
```

ianhi commented:
Interesting how we have different perspectives on zarr usage from using it in different contexts!


So is this URL meant to be used as an HTTP store? If so, why is it using S3 infrastructure to achieve that, can that be easily changed?

> In your example, I see that you get a warning when you do this at the end:


I listed all the objects with : `aws s3 ls s3://idr/zarr/v0.5/idr0062A/6001240_labels.zarr/ --endpoint-url  https://uk1s3.embassy.ebi.ac.uk/ --no-sign-request --recursive`

and that error is coming from this line: https://github.com/zarr-developers/zarr-python/blob/be519b006f3073c3c0d56801cd2daa597538fd59/src/zarr/core/group.py#L3535-L3549

due to this extra object in the hierarchy:

`zarr/v0.5/idr0062A/6001240_labels.zarr/6001240_labels.zarr/ro-crate-metadata.json`

ianhi commented:
> Also, it would be nice to enable dialog closing when you click on the background.

ahhh yes good call. I hate it when modals don't do that, can't believe i didn't think of that.


> I tried this again...


gahh. that is so bizarre. I've never seen that before. This is after just following the instructions here? https://pixi.sh/dev/installation/ I wonder if it's that your pixi install is somewhat broken, or if somehow the ruby bundler is trying to access other areas

ianhi committed:
clicking outside modal closes it

Co-authored-by: William Moore <w.moore@dundee.ac.uk>

will-moore commented:
Seems that the problem is that I don't have permissions to write to my default `$TMPDIR` location.

```
echo $TMPDIR
/var/folders/zz/zyxvpxvq6csfxvn_n0000000000000/T/

$ ls $TMPDIR
ls: /var/folders/zz/zyxvpxvq6csfxvn_n0000000000000/T/: Permission denied
```
I also don't have `sudo` permissions on my Unversity-administered laptop. But I can at least run the commands with that work-around so that's fine for now.

ianhi committed:
tmp dir instructions

ianhi committed:
add https url + some formatting and documentaiton for user

ianhi commented:
Changes:

1. added HTTPS URL to the modal
2. Updated the docs at the top from just aws download to poitning users to the button: 
<img width="841" height="161" alt="image" src="https://github.com/user-attachments/assets/996302ce-f834-4ae7-b9dc-28f673aee5ec" />
3. grouped each section with an outline to make it easy to see that the two copy links for S3 URLs are the grouped. Added to everything so it didn't look weird.
<img width="718" height="778" alt="image" src="https://github.com/user-attachments/assets/28ae9f2d-a94f-4bd9-9bfe-abdde9f8bd19" />

ianhi committed:
ordering

will-moore reviewed (approved):
Testing locally, with main merged in. Looks great, thanks.

@joshmoore you OK to merge (I don't have permissions)?

will-moore commented:
@ianhi could you merge origin/main into your branch so we can see the changes here a bit clearer? Thanks.

joshmoore committed:
Merge remote-tracking branch 'origin/main' into fix-url

joshmoore commented:
I merged in main. Looks good. Thanks, all!

ianhi commented:
thanks both!
